InventoryLowQuantityNotificationAdminUi/Observer/ProcessSourceItemConfigurationsObserver.php

```diff
@@ -95,11 +95,49 @@ public function execute(EventObserver $observer)
             ];
         } else {
             $sources = $controller->getRequest()->getParam('sources', []);
-            if (isset($sources['assigned_sources']) && is_array($sources['assigned_sources'])) {
-                $assignedSources = $sources['assigned_sources'];
-            }
+            $stockData = $controller->getRequest()->getParam('product', [])['stock_data'] ?? [];
+            $assignedSources =
+                isset($sources['assigned_sources'])
+                && is_array($sources['assigned_sources'])
+                    ? $this->updateAssignedSources($sources['assigned_sources'], $stockData)
+                    : [];
         }
 
         $this->sourceItemsConfigurationProcessor->process($product->getSku(), $assignedSources);
     }
+
+    /**
+     * Update assign source item quantity, status notify_stock_qty and notify_stock_qty_use_default
+     *
+     * @param array $assignedSources
+     * @param array $stockData
+     * @return array
+     */
+    private function updateAssignedSources(array $assignedSources, array $stockData): array
+    {
+        foreach ($assignedSources as $key => $source) {
+            $isSourceUpdated = false;
+            if (!key_exists('quantity', $source) && isset($source['qty'])) {
+                $source['quantity'] = (int) $source['qty'];
```

Comment:
Please, use `$assignedSources[$key]['quantity']` instead and remove `$isSourceUpdated` flag usage in all cases.

Reply:
Redundant code refactored